eZ/Publish/Core/Persistence/Legacy/Content/UrlWildcard/Gateway/DoctrineDatabase.php

```diff
@@ -162,4 +163,36 @@ public function loadUrlWildcardsData($offset = 0, $limit = -1)
 
         return $stmt->fetchAll(\PDO::FETCH_ASSOC);
     }
+
+    /**
+     * Loads the UrlWildcard with given $sourceUrl.
+     *
+     * @param string $sourceUrl
+     *
+     * @return array
+     */
+    public function loadUrlWildcardBySourceUrl(string $sourceUrl): array
+    {
+        /** @var \Doctrine\DBAL\Connection $connection */
+        $connection = $this->dbHandler->getConnection();
+        $queryBuilder = $connection->createQueryBuilder();
+        $expr = $queryBuilder->expr();
+        $queryBuilder->select(
+            'id',
+            'destination_url',
+            'source_url',
+            'type'
+        )
+        ->from('ezurlwildcard')
+        ->where(
+            $expr->eq(
+                'source_url',
+                $queryBuilder->createNamedParameter($sourceUrl)
+            )
+        );
+
+        $result = $queryBuilder->execute()->fetch(FetchMode::ASSOCIATIVE);
+
+        return $result ?: [];
```

Comment:
Can we use here:
```suggestion
        return $result ?? [];
```
?

Reply:
Unfortunately, not. This method is supposed to return `array` in all cases, but `$queryBuilder->execute()->fetch()` may return `false`, which is consider as correct for *null* coalescing operator. See: https://3v4l.org/nQOaR

Reply:
Ah, that's true. I'm in such cases fan of something more explicit like 
```php
$result !== false ? $result : []
```` 
but what you have here is good anyway :+1: